contract/models/contract.py

```diff
@@ -123,6 +123,41 @@ class ContractContract(models.Model):
         copy=False,
         track_visibility="onchange",
     )
+    modification_ids = fields.One2many(
+        comodel_name='contract.modification',
+        inverse_name='contract_id',
+        string='Modifications',
+    )
+
+    @api.multi
+    def write(self, vals):
+        res = super(ContractContract, self).write(vals)
+        if 'modification_ids' in vals:
+            self._modification_mail_send()
+        return res
+
+    @api.model
+    def _modification_mail_send(self):
+        self.ensure_one()
+        modification_ids_not_notify = self.modification_ids.filtered(
+            lambda x: not x.mail_notify
+        )
+        if modification_ids_not_notify:
+            template = self.env.ref(
+                'contract.email_contract_modification_template', False
```

Comment:
You can remove the `False` for not silently fail on this.

Reply:
I'm thinking of using QWeb template instead of this for several reasons:

- Being more hidden from modifications.
- Allow inheritance.

Anyway, the message should be put in the thread. There's an option for using QWeb from mail templates as an envelope, but maybe we can trick it for being the full template and empty inner body.

What do you think?

Reply:
IMO QWeb template it's a good idea. I will change it now.